### ISSUE TITLE
Inherited settings for JNLP launcher not saved

### ISSUE THREAD

**Issue by earvidso** [jenkinsci/docker-plugin#411]
If "Launch method" is set to "Docker JNLP launcher", the inherited settings "Tunnel connection through" and "JVM options" can be set but they are not saved. At the moment I have a TCP tunnel set up on the reverse proxy itself but would like to get rid of it and use randomized ports whenever that may be possible.

**Comment by earvidso** [comment]
Also this happens when one of the JNLP computers is disconnected (not a big deal, and I might be able to fix it myself):

Aug 06, 2016 3:28:47 AM hudson.slaves.SlaveComputer$2 onClosed
SEVERE: Launcher com.nirima.jenkins.plugins.docker.launcher.DockerComputerJNLPLauncher@5c1d6e26s afterDisconnect method propagated an exception when {1}s connection was closed: Launcher must not be null
java.lang.IllegalStateException: Launcher must not be null
    at com.nirima.jenkins.plugins.docker.launcher.DockerComputerLauncher.getLauncher(DockerComputerLauncher.java:54)
    at com.nirima.jenkins.plugins.docker.launcher.DockerComputerLauncher.afterDisconnect(DockerComputerLauncher.java:71)
    at hudson.slaves.SlaveComputer$2.onClosed(SlaveComputer.java:503)
    at hudson.remoting.Channel.terminate(Channel.java:868)
    at org.jenkinsci.remoting.nio.NioChannelHub$3.run(NioChannelHub.java:614)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
    at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Aug 06, 2016 3:28:47 AM jenkins.slaves.JnlpSlaveAgentProtocol$Handler$1 onClosed
WARNING: Computer.threadPoolForRemoting [#63] for build-e41b9051d4a5 terminated
java.io.EOFException
    at org.jenkinsci.remoting.nio.NioChannelHub$3.run(NioChannelHub.java:614)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
    at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)

**Comment by carlossg** [comment]
same here, the logs have this stack, so it seems related to #432

```
falling back to default instantiation com.nirima.jenkins.plugins.docker.launcher.DockerComputerJNLPLauncher {"stapler-class":"com.nirima.jenkins.plugins.docker.launcher.DockerComputerJNLPLauncher","user":"","tunnel":"10.0.0.102:50000","vmargs":""}
java.lang.NullPointerException
	at hudson.model.Descriptor$NewInstanceBindInterceptor.instantiate(Descriptor.java:641)
	at org.kohsuke.stapler.RequestImpl.instantiate(RequestImpl.java:749)
	at org.kohsuke.stapler.RequestImpl.access$200(RequestImpl.java:83)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:674)
	at org.kohsuke.stapler.RequestImpl.bindJSON(RequestImpl.java:478)
	at org.kohsuke.stapler.RequestImpl.injectSetters(RequestImpl.java:827)
	at org.kohsuke.stapler.RequestImpl.instantiate(RequestImpl.java:777)
	at org.kohsuke.stapler.RequestImpl.access$200(RequestImpl.java:83)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:674)
	at org.kohsuke.stapler.RequestImpl.bindJSON(RequestImpl.java:478)
	at org.kohsuke.stapler.RequestImpl.bindJSON(RequestImpl.java:474)
	at hudson.model.Descriptor.newInstance(Descriptor.java:590)
	at hudson.model.Descriptor$NewInstanceBindInterceptor.onConvert(Descriptor.java:661)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:617)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:712)
	at org.kohsuke.stapler.RequestImpl.bindJSON(RequestImpl.java:478)
```

**Comment by ndeloof** [comment]
fixed in 0.17+
